File: testers/avro/java/generated-and-checked-in/com/example/events/Email.java
package com.example.events;

/** Customer email address */
public record Email(String value) {
  public Email withValue(String value) {
    return new Email(value);
  }

  @Override
  public java.lang.String toString() {
    return value.toString();
  }

  /** Create a Email from a raw value */
  public static Email valueOf(String v) {
    return new Email(v);
  }

  /** Get the underlying value */
  public String unwrap() {
    return this.value();
  }
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/Email.kt
package com.example.events



/** Customer email address */
data class Email(val value: kotlin.String) {
  /** Get the underlying value */
  fun unwrap(): kotlin.String {
    return this.value
  }

  override fun toString(): kotlin.String {
    return value
  }

  companion object {
    /** Create a Email from a raw value */
    fun valueOf(v: kotlin.String): Email {
      return Email(v)
    }
  }
}